src/frequenz/sdk/actor/_actor.py

```diff
@@ -44,13 +44,33 @@ class Actor(BackgroundService, abc.ABC):
         This is mostly used for testing purposes and shouldn't be set in production.
     """
 
+    def __init__(self, *, name: str | None = None) -> None:
+        """Create actor instance.
+
+        Args:
+            name: The name of this background service.
+        """
+        super().__init__(name=name)
+        self._is_cancelled = False
+
     def start(self) -> None:
         """Start this actor.
 
         If this actor is already running, this method does nothing.
+
+        Raises:
+            RuntimeError: If actor was canceled. Canceled actor can't
+                be started again.
         """
         if self.is_running:
             return
+
+        if self._is_cancelled:
+            raise RuntimeError(
+                "Actor %s: was canceled and can't be re-run."
+                "Please create another actor instance.",
+            )
```

Comment:
I wouldn't go for this at the actor level yet. We have many actors that assume they can be restarted and count on this. I would like to make a big overhaul of how we handle tasks and actors, I would avoid making any breaking changes until then as adapting all existing code to any breaking changes here will be very costly.